lib/src/data/repository/my_local_repository.dart

```diff
@@ -9,32 +12,58 @@ class MyLocalRepository implements IMyRepository {
   static const String genresAsset = 'assets/genres.json';
 
   @override
-  Future<List<MovieModel>> loadMovies() async {
+  Future<DataState<List<MovieModel>>> loadMovies() async {
     final String response = await rootBundle.loadString(moviesAsset);
-    final data = await json.decode(response);
-    return MovieModel.listOfMovies(data);
+    final List<dynamic> data = await json.decode(response);
+    return DataSuccess<List<MovieModel>>(data: MovieModel.listOfMovies(data));
   }
 
   @override
-  Future<List<GenreModel>> loadGenres() async {
-    List<GenreModel> genres = [];
+  Future<DataState<List<GenreModel>>> loadGenres() async {
+    List<GenreModel> genres = <GenreModel>[];
     final String response = await rootBundle.loadString(genresAsset);
-    final data = await json.decode(response);
+    final Map<String, dynamic> data = await json.decode(response);
     data['genres'].forEach(
-      (genreJson) {
+      (Map<String, dynamic> genreJson) {
         GenreModel genre = GenreModel.fromJson(genreJson);
         genres.add(genre);
       },
     );
-    return genres;
+    return DataSuccess<List<GenreModel>>(data: genres);
   }
 
   Future<List<String>> getGenresById(List<int> genres) async {
-    final List<GenreModel> genresJson = await loadGenres();
-    List<String> genreNames = genresJson
-        .where((genreJson) => genres.contains(genreJson.id))
-        .map((genreJson) => genreJson.name)
+    final DataState<List<GenreModel>> genresJson = await loadGenres();
+    List<String> genreNames = genresJson.data
+        !.where((GenreModel genreJson) => genres.contains(genreJson.id))
+        .map((GenreModel genreJson) => genreJson.name)
         .toList();
     return genreNames;
   }
+
+  MovieModel get getMockMovie => MovieModel.mockMovie();
+
+  @override
+  Future<DataState<List<Movie>>> loadPopularMovies() {
+    // TODO: implement loadPopularMovies
+    return loadMovies();
+  }
+
+  @override
+  Future<DataState<List<Movie>>> loadNowPlayingMovies() {
+    // TODO: implement loadPlayingNowMovies
+    return loadMovies();
+  }
+
+  @override
+  Future<DataState<List<Movie>>> loadUpcomingMovies() {
+    // TODO: implement loadUpcomingMovies
+    return loadMovies();
+  }
+
+  @override
+  Future<DataState<List<Cast>>> loadCast(int movieId) {
+    throw UnimplementedError();
+  }
```

Comment:
why all this methods call loadMovies, and also why you are overriding the loadCast call?

Reply:
That because i define it in the interface. Now i have the contrats of 
```dart
loadMoviesByType(Endpoint endpoint);
loadGenres();
loadCast(int movieId);
```
So i have to implement it in the locar repository but it hasn't any logic at all, so i call the main method (in this case `loadMovies()` )

Reply:
Do you mean to have the `loadGenres` in a mixin that only the remote repository implement it? 
As i see it the problem there is that i use the abstract class `IMyRepository` in the use cases. What should i do there? Because if insted of having `IMyRepository` i do `APIRepository` the abstract class is no use at all, and it has no expansibility (for example to call a database).
What do you think abaout this? hehe